From c0f63850ad29ec978d070a08b816dc2bfca337e3 Mon Sep 17 00:00:00 2001
From: Gustavo Sverzut Barbieri <barbieri@profusion.mobi>
Date: Fri, 30 Dec 2016 11:28:41 -0200
Subject: [PATCH] udev-builtin-input_id.c: add missing kernel header defines.

Add missing defines so eudev builds for older kernels such as 2.6

Signed-off-by: Gustavo Sverzut Barbieri <barbieri@profusion.mobi>
---
 src/shared/missing.h             | 8 ++++++++
 src/udev/udev-builtin-input_id.c | 1 +
 2 files changed, 9 insertions(+)

diff --git a/src/shared/missing.h b/src/shared/missing.h
index 5ad599795..bebbb42be 100644
--- a/src/shared/missing.h
+++ b/src/shared/missing.h
@@ -171,3 +171,11 @@ static inline int name_to_handle_at(int fd, const char *name, struct file_handle
     (char *)memcpy(__new, __old, __len); \
   })
 #endif
+
+#ifndef BTN_TRIGGER_HAPPY
+#define BTN_TRIGGER_HAPPY 0x2c0
+#endif
+
+#ifndef INPUT_PROP_MAX
+#define INPUT_PROP_MAX 0x1f
+#endif
diff --git a/src/udev/udev-builtin-input_id.c b/src/udev/udev-builtin-input_id.c
index b14190e42..ca545be5d 100644
--- a/src/udev/udev-builtin-input_id.c
+++ b/src/udev/udev-builtin-input_id.c
@@ -32,6 +32,7 @@
 
 #include "udev.h"
 #include "util.h"
+#include "missing.h"
 
 /* we must use this kernel-compatible implementation */
 #define BITS_PER_LONG (sizeof(unsigned long) * 8)
-- 
2.11.0

